### PR TITLE
In-cluster problem writer

### PR DESCRIPTION
**Pull request by bomoko**
This PR replaces the entire current fact writing subsystem with [in-cluster problems writing](https://github.com/uselagoon/insights-remote?tab=readme-ov-file#insights-written-directly-to-insights-remote).

As such, it removes almost all of the API connections as well as all references to Facts.

However, there is still one direct connection to the Lagoon GraphQL API in cases where there are no problems registered, and all problems are to be cleared. This can be replaced in a (near) future version when https://github.com/uselagoon/insights-remote/pull/18 is released (potentially in Lagoon 2.18).

UPDATE, 2 Feb 2024 - It struck me as important that the only reason that the lagoon _formatter_ existed was because of the [Lagoon Facts App](https://github.com/uselagoon/lagoon-facts-app/). Having a specific formatter for Lagoon seems unimportant when you're writing the facts directly, so this PR now removes the formatter, and just has the pushing of problems to Lagoon as an optional switch that can be used with whichever formatter.

### CODE REVIEW

File: pkg/shipshape/shipshape.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 
 var RunConfig config.Config
 var RunResultList result.ResultList
-var OutputFormats = []string{"json", "junit", "simple", "table", "lagoon-facts"}
+var OutputFormats = []string{"json", "junit", "simple", "table", "lagoon-problems"}
```
Comment (yusufhm) [inline pkg/shipshape/shipshape.go:22]: I believe this is not required anymore.
```suggestion
var OutputFormats = []string{"json", "junit", "simple", "table"}
```

File: pkg/lagoon/lagoon.go [hunk 2]
```diff
@@ -20,12 +25,31 @@ type Fact struct {
...
+	AssociatedPackage string                `json:"associatedPackage,omitempty"`
+	Description       string                `json:"description,omitempty"`
+	Links             string                `json:"links,omitempty"`
+}
+
 const SourceName string = "Shipshape"
 const FactMaxValueLength int = 300
 
 var ApiBaseUrl string
 var ApiToken string
```
Comment (yusufhm) [inline pkg/lagoon/lagoon.go:49]: Are we still using these given we're now using the in-cluster token?

**Review by yusufhm — CHANGES_REQUESTED** [review]
Great work @bomoko - I just left a couple comments regarding some potentially obsolete pieces of code now.

File: pkg/lagoon/lagoon.go (hunk 2 above)
Comment (bomoko) [reply pkg/lagoon/lagoon.go:49]: Yep, just until the release of Lagoon 2.18, really - for clearing problems when none are found.

Comment (yusufhm) [reply pkg/lagoon/lagoon.go:49]: Oh yes, you mentioned that

File: main.go [hunk 3]
```diff
@@ -80,7 +80,7 @@ func main() {
 
 	determineLogLevel()
 
-	if outputFormat == "lagoon-facts" && lagoon.PushFacts {
+	if outputFormat == "lagoon-problems" && lagoon.PushProblems {
```
[the remark this answers is not in the document]
Comment (bomoko) [reply main.go:83]: Good catch - I think the only reason I changed it was a find/replace I did for "facts/problems".
The change you suggest is right, Once 2.18 is out, we can kill this block

**Review by yusufhm — APPROVED** [review]
LGTM

**Pull request merged**